eval/adjust.rkt

```diff
@@ -78,29 +78,30 @@
     (define srcs (map (lambda (x) (vector-ref vregs x)) tail-registers)) ; tail of the current instr
     (define output (vector-ref vregs n))                            ; output of the current instr
     
-    (define ampls-from-above (vector-ref vprecs-new (- n varc)))    ; vprecs-new is shifted by varc elements from vregs
-    (define ampls (get-ampls op output srcs))
+    (define intro (vector-ref vprecs-new (- n varc)))               ; intro for the current instruction
+    (define ampls (get-ampls op output srcs))                       ; ampls for the tail instructions
 
-    (define final-parent-precision (max (+ ampls-from-above
+    (define final-parent-precision (max (+ intro
```

Comment:
This can be changed to exactly what we have mathematically as:
`max(2, p[z_i] + 2 + log_2[intro]`

Reply:
Yeah. `log_2[intro]` is just `logspan(z)`, so this would just be `max(2, intro + 2 + logspan(output))`